litex_boards/platforms/sipeed_tang_nano_9k.py

```diff
@@ -99,7 +99,7 @@ class Platform(GowinPlatform):
     default_clk_period = 1e9/27e6
 
     def __init__(self, toolchain="gowin"):
-        GowinPlatform.__init__(self, "GW1NR-LV9QN88PC6/I5", _io, _connectors, toolchain=toolchain, devicename="GW1NR-9C")
+        GowinPlatform.__init__(self, "GW1NR-LV9QN88PC6/I5", _io, _connectors, toolchain=toolchain, devicename="GW1N-9C")
```

Comment:
This fixes the `--family` passed to Nextpnr, but might break the Gowin toolchain. Unclear what the correct option is here. Gowin is notoriously ambiguous and inconsistent with their part names.

@yrabbit what do you think about just making a symlink in nextpnr for the R devices? This is a problem people keep running into because it's just extremely confusing that the `C` is nowhere in the part number and the `R` isn't part of the database name.

Reply:
What do you think to use something like https://github.com/enjoy-digital/litex/blob/master/litex/build/xilinx/f4pga.py#L56: this allows to have this change only once is the appropriate module?

Reply:
That would be one solution but we get occasional bug reports from people who pass a wrong family or whatever and it doesn't work and Gowin part numbers are really messed up so having some safety net in Apicula / Nextpnr is preferrable I think.

Basically, there is a regex that parses the long part number, and a family argument. You'd think the long one is more specific but no, the C can only be seen in the date code. So sometimes you need both or to just specify the family. But the mapping from part to family is confusing. So that's why I'm suggesting some fallbacks for providing too much information in the family. For example R is not a separate family.

Reply:
I guess the alternative is to filter the family in code, stripping elements that don't translate to a database

Reply:
I ended up just doing what trabucayre suggested